### PR TITLE
Detpack Buff

### PR DESCRIPTION
**Pull request by novaepee**
## About The Pull Request

Adding the following:

1) detpack explosive in req now cost 10 points and have 6 detpacks
2) a LOT of detpack in req, which costs 15 points and 12 detpacks
3) new squad engineer detpack pouch which holds 6 detpacks
4) said detpack pouch is in essential set
5) reduce price of individual detpack in squad engineer

## Why It's Good For The Game

Araulius wanted it, and when I saw how much detpacks squad engineers have, they don't even spawn with one compared to a C4. Also, release the ordnance nature of squad engineer. Total balance, I swear.

## Changelog
:cl:
add: A lot of detpacks
expansion: Give more detpacks in req
qol: Let squad engineer have more detpacks by letting them spawn in with detpack
balance: A lot of detpacks
/:cl:

### CODE REVIEW

**Comment by EvilMelonTheSecond** [comment]
Detpacks suck, no amount of discount or storage buff will make them see use. Claymores are superior in every way so this feels like adding a noob trap.

File: code/game/objects/items/storage/pouch.dm [hunk 1]
```diff
@@ -320,6 +320,24 @@
 		/obj/item/ammo_magazine/standard_smartmachinegun,
 	)
 
+/obj/item/storage/pouch/explosive/engineerdetpack/Initialize()
```
Comment (hyper2snyper) [inline code/game/objects/items/storage/pouch.dm:323]: XD This is so bad.

Ok so, when you make a proc for a typepath that isnt defined, it creates the typepath regardless. It just inherits from the parent.

So basically you dont want all these definitions inside the Initialize() for goodness sakes.

File: code/modules/reqs/supplypacks.dm [hunk 2]
```diff
@@ -507,6 +507,29 @@ EXPLOSIVES
...
+		/obj/item/detpack,
+		/obj/item/detpack,
+		/obj/item/detpack,
+		/obj/item/detpack,
+		/obj/item/detpack,
+		/obj/item/assembly/signaler,
+	)
+	cost = 10
+
+/datum/supply_packs/explosives/detpack
```
Comment (hyper2snyper) [inline code/modules/reqs/supplypacks.dm:520]: this is a duplicate definition of the one above it.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

**Comment by AndroidMK2** [comment]
This doesn't solve the problem, being able to have more of currently useless explosive ain't really changing much, better buff the detpacks themselves.

File: code/game/objects/items/storage/pouch.dm (hunk 1 above)
Comment (novaepee) [reply code/game/objects/items/storage/pouch.dm:323]: Yeah, I'm not going to bother making a new pouch.

File: code/modules/reqs/supplypacks.dm (hunk 2 above)
Comment (novaepee) [reply code/modules/reqs/supplypacks.dm:520]: Intended. It's to have multiple detpack. Req code is wack.

Comment (hyper2snyper) [inline code/modules/reqs/supplypacks.dm:520]: this is still a duplicate declaration.
look 13 lines above, at line 507, you can see the declaration there is the exact same as the one you typed. This means that the second declaration (the one at line 520) will be the only one to exist and the first one (line 507) wont matter anymore

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/pouch.dm [hunk 3]
```diff
@@ -359,6 +358,19 @@
 	fill_type = /obj/item/explosive/grenade/upp
 	fill_number = 4
 
+/obj/item/storage/pouch/detpack
```
Comment (BraveMole) [inline code/game/objects/items/storage/pouch.dm:361]: no, as any pouch it should be selectable by engie AND starts full, and/or be available in surplus vendor and be empty

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 4]
```diff
@@ -654,6 +654,8 @@
 /obj/effect/essentials_set/engi
 	spawned_gear_list = list(
 		/obj/item/explosive/plastique,
+		/obj/item/storage/pouch/detpack,
+		/obj/item/assembly/signaler,
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:658]: no, engie has enough shit already in essentials set

File: code/modules/reqs/supplypacks.dm [hunk 5]
```diff
@@ -507,6 +507,29 @@ EXPLOSIVES
 /datum/supply_packs/explosives/detpack
```
Comment (BraveMole) [inline code/modules/reqs/supplypacks.dm:507]: no one supply pack is enough

**Review by BraveMole — CHANGES_REQUESTED** [review]
I doubt this does anything. maybe buff detpacks, at least against mazes walls?

**Comment by novaepee** [comment]
Sorry, Araulius, not worth.

**Pull request closed without merging**